glommio/src/executor.rs

```diff
@@ -781,15 +753,21 @@ impl LocalExecutor {
     /// });
     /// ```
     pub fn spawn<T: 'static>(&self, future: impl Future<Output = T> + 'static) -> Task<T> {
-        let ex = self
+        let tq = self
             .queues
             .borrow()
             .active_executing
             .as_ref()
-            .map(|x| x.borrow().ex.clone())
-            .or_else(|| self.get_executor(&TaskQueueHandle { index: 0 }))
-            .unwrap();
-        Task(ex.spawn(future))
+            .map_or_else(
+                || self.get_queue(&TaskQueueHandle { index: 0 }),
+                |x| Some(x).cloned(),
```

Comment:
```suggestion
                |x| Some(x.clone()),
```

Reply:
Ah, I didn't realise that `x` is a `Rc`... so it should be `|x| Some((*x).clone())` instead. Sorry.